src/Sylius/Bundle/ApiBundle/CommandHandler/AddProductReviewHandler.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\CommandHandler;
+
+use Sylius\Bundle\AdminApiBundle\Model\UserInterface;
+use Sylius\Bundle\ApiBundle\Command\AddProductReview;
+use Sylius\Bundle\ApiBundle\Context\UserContextInterface;
+use Sylius\Bundle\ApiBundle\Provider\CustomerProviderInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Sylius\Component\Core\Repository\ProductRepositoryInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\Review\Model\ReviewInterface;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+
+/** @experimental */
+final class AddProductReviewHandler implements MessageHandlerInterface
+{
+    /** @var CustomerProviderInterface */
+    private $customerProvider;
+
+    /** @var RepositoryInterface */
+    private $productReviewRepository;
+
+    /** @var FactoryInterface */
+    private $productReviewFactory;
+
+    /** @var ProductRepositoryInterface */
+    private $productRepository;
+
+    public function __construct(
+        CustomerProviderInterface $customerProvider,
+        RepositoryInterface $productReviewRepository,
+        FactoryInterface $productReviewFactory,
+        ProductRepositoryInterface $productRepository
+    ) {
+        $this->customerProvider = $customerProvider;
+        $this->productReviewRepository = $productReviewRepository;
+        $this->productReviewFactory = $productReviewFactory;
+        $this->productRepository = $productRepository;
+    }
+
+    public function __invoke(AddProductReview $addProductReview): ReviewInterface
```

Comment:
I'm thinking about this return type. I just checked few handlers and all of them returns void. So is it really needed?

Reply:
Command `AddProductReview` create new `ReviewInterface`. Its useful to return `ReviewInterface` as a `Response`.

Reply:
it is needed if we want to return 201 and created object. But not sure if this is case here

Reply:
If I add void, As a response I get object of `AddProductReview` and `201` code. Its needed? Do user should know structure of this object? This is the intention creating `ReviewInterface` object and it should return this object as a response. IMO